internal/server/authz/middleware/grpc/middleware.go

```diff
@@ -25,14 +25,32 @@ type InterceptorOptions struct {
 	skippedServers []any
 }
 
-func skipped(ctx context.Context, server any, o InterceptorOptions) bool {
-	if skipSrv, ok := server.(SkipsAuthorizationServer); ok && skipSrv.SkipsAuthorization(ctx) {
+var (
+	// methods which should always skip authorization
+	skippedMethods = map[string]any{
+		"/flipt.auth.AuthenticationService/GetAuthenticationSelf":    struct{}{},
+		"/flipt.auth.AuthenticationService/ExpireAuthenticationSelf": struct{}{},
+	}
+)
+
+func skipped(ctx context.Context, info *grpc.UnaryServerInfo, o InterceptorOptions) bool {
+	// if we skip authentication then we must skip authorization
+	if skipSrv, ok := info.Server.(authmiddlewaregrpc.SkipsAuthenticationServer); ok && skipSrv.SkipsAuthentication(ctx) {
+		return true
+	}
+
+	if skipSrv, ok := info.Server.(SkipsAuthorizationServer); ok && skipSrv.SkipsAuthorization(ctx) {
+		return true
+	}
```

Comment:
Just checking, is this still necessary? I see you removed a lot of the SkipsAuthorization methods, but I can't tell if its all of them.

Reply:
so the only places where `SkipsAuthorization` is now used are:

1. EvaluationServer - where we still want authn but not authz (every valid user should be able to evaluate)
2. MetadataServer - required for UI to work, includes config and info endpoints
3. GRPCHealthcheck endpoint 

I think we do still need `SkipsAuthorization` for these

Reply:
Nice, makes sense for now 👍 

I think you could make a case for each of those services having a slightly different behaviour than skip authz though going forward.

- Evaluation Server

Folks probably would eventually want to say this thing can e.g. only evaluate (like a token or service account in k8s).
In which case we likely need an `evaluate` verb on the `flag` resource.

- Metadata Server

Im sure there will be a desire for only e.g. admins being able to access (certain?) service configuration.
But for now skipping it makes sense.

- GRPCHealthcheck

I think this should skip authn right, so it should short-circuit before authz? As healthcheck systems rarely go through authn paths at all. At best you might run them on a separate port that is not accessible outside your internal network to protect them.

All this said though, I think we might need to say SkipAuthorization should be checked independently of SkipAuthentication. Right now its in the compound `skipped` function, but we have cases like `MetadataServer` where you at-least want authentication to be performed and not skipped.
Whereas the current implementation will skip authn because it reports authz as skipped.